tensorflow/dtensor/mlir/utils/collective_lowering.cc

```diff
@@ -506,23 +620,44 @@ mlir::LogicalResult LowerReduceScatterOp(
             reduce_scatter.getReduceOpAttr());
     SetSingleLayoutOnOp(xla_reduce_scatter, *output_layout);
     reduce_scatter.replaceAllUsesWith(xla_reduce_scatter);
+  } else if (reduce_scatter.getDeviceType().endswith("GPU")) {
```

Comment:
I recall the new Op only work for NCCL GPU? In that case we may want to protect this branch with the 'DTENSOR_GPU_USE_NCCL_COMMUNICATION' environment variable, which is used here:

https://github.com/tensorflow/tensorflow/blob/0c1163404de5246ebc0f4366dd079e46c0dea95d/tensorflow/dtensor/python/config.py

It is unclear to me how to best ensure the C++ and Py definition of the config environment is consistent, but for now we can leave this as a debt and just at the top of this file something like https://github.com/tensorflow/tensorflow/blob/0c1163404de5246ebc0f4366dd079e46c0dea95d/tensorflow/dtensor/mlir/utils/collective_lowering.cc#L71